src/main/java/dev/jbang/Util.java

```diff
@@ -76,8 +82,36 @@ static public void writeString(Path toPath, String scriptText) throws IOExceptio
 	private static final Pattern mainClassPattern = Pattern.compile(
 			"(?sm)class *(\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*) .*static void main");
 
+	public static OS getOS() {
+		String os = System.getProperty("os.name").toLowerCase(Locale.ENGLISH).replaceAll("[^a-z0-9]+", "");
+		if (os.startsWith("mac") || os.startsWith("osx")) {
+			return OS.mac;
+		} else if (os.startsWith("linux")) {
+			return OS.linux;
+		} else if (os.startsWith("win")) {
+			return OS.windows;
+		} else {
+			throw new ExitException(CommandLine.ExitCode.SOFTWARE, "Unsupported OS: " + os);
+		}
+	}
+
+	public static Arch getArch() {
+		String arch = System.getProperty("os.arch").toLowerCase(Locale.ENGLISH).replaceAll("[^a-z0-9]+", "");
+		if (arch.matches("^(x8664|amd64|ia32e|em64t|x64)$")) {
+			return Arch.x64;
+		} else if (arch.matches("^(x8632|x86|i[3-6]86|ia32|x32)$")) {
+			return Arch.x32;
+		} else {
+			throw new ExitException(CommandLine.ExitCode.SOFTWARE, "Unsupported architecture: " + arch);
+		}
+	}
+
 	public static boolean isWindows() {
-		return System.getProperty("os.name").toLowerCase().contains("windows");
+		return getOS() == OS.windows;
+	}
+
+	public static boolean isMac() {
+		return getOS() == OS.windows;
```

Comment:
this aint right ;)

Reply:
```suggestion
		return getOS() == OS.mac;
```

fixing that then download for mac works

Reply:
Oops hehe